src/main/scala/is/hail/expr/Type.scala

```diff
@@ -1023,6 +1035,18 @@ case class TVariant(gr: GRBase) extends ComplexType {
 
   override val partitionKey: Type = TLocus(gr)
 
+  override def typedOrderedKey[PK, K]: OrderedKey[PK, K] = new OrderedKey[PK, K] {
+    def project(key: K): PK = key.asInstanceOf[Variant].locus.asInstanceOf[PK]
```

Comment:
I have little faith left in trying to use the Scala type system, but I've ran into situations like this as well where I wanted to know that `TVariant` was the type for `Variant`.

Reply:
Yeah.  For a while I would been for Type[T], but this is really temporary to support the old "triple" format until we switch everything to RegionValues.